index.js

```diff
@@ -153,7 +153,18 @@ function user_check_password(password, hashedPassword) {
     default:
       return false;
   }
-  return (hashed && stored_hash == hashed);
+
+  // Use a constant time comparison to prevent timing attacks.
+  if (hashed) {
+    var mismatch = 0;
+    for (var i = 0, l = hashed.length; i < l; ++i) {
```

Comment:
We need to also check that the strings are the same length. Here’s an example. https://github.com/hueniverse/hawk/commit/c6d1a123cba27c6dc94229d89fa607084999e715#diff-6d186b954a58d5bb740f73d84fe39073R279